diff --git a/fesvr/dtm.cc b/fesvr/dtm.cc
index 993011d..96a67ba 100644
--- a/fesvr/dtm.cc
+++ b/fesvr/dtm.cc
@@ -6,6 +6,7 @@
 #include <string.h>
 #include <assert.h>
 #include <pthread.h>
+#include <stdexcept>
 
 #define RV_X(x, s, n) \
   (((x) >> (s)) & ((1 << (n)) - 1))
diff --git a/riscv/devices.h b/riscv/devices.h
index 1bc9618..3dd6c66 100644
--- a/riscv/devices.h
+++ b/riscv/devices.h
@@ -7,6 +7,7 @@
 #include <string>
 #include <map>
 #include <vector>
+#include <stdexcept>
 
 class processor_t;
 
